src/core/public/application/application_service.tsx

```diff
@@ -392,3 +394,12 @@ const updateStatus = (app: App, statusUpdaters: AppUpdaterWrapper[]): App => {
     ...changes,
   };
 };
+
+const populateDeepLinkDefaults = (deepLinks?: AppDeepLink[]): AppDeepLink[] => {
+  if (!deepLinks) return [];
+  return deepLinks.map((deepLink) => ({
+    ...deepLink,
+    navLinkStatus: deepLink.navLinkStatus ?? AppNavLinkStatus.default,
+    deepLinks: populateDeepLinkDefaults(deepLink.deepLinks),
```

Comment:
Do we need to do this for the nested deep links? If the UI isn't going to support displaying the nested links, I think the API should reflect that. Maybe the `deepLinks` property of `AppDeepLink` should be something like `deepLinks?: Omit<AppDeepLink, 'navLinkStatus'>[];`

However, if it's simpler to just support displaying nested links in the UI too, then we can leave as-is.

Reply:
Interesting point.

Currently, I am implementing it to support nested deepLinks in the navigation, so all deepLinks with `path` and `navLinkStatus: visible` will be displayed, all relative to the main `appRoute` regardless of the depth inside the deepLinks entry.

I did it this way because it is possible to have "logical" sections inside deepLinks without path, just for hierarchization purposes, so to give more freedom I allowed the nested deepLinks to be displayed normally. They will always have control of what will be displayed using/omitting the `navLinkStatus`.

Reply:
That works for me, the logical sections argument seems like a good justification